force-app/main/default/classes/FuzzyPicklists.cls

```diff
@@ -103,20 +105,18 @@ public class FuzzyPicklists {
     
     public static String buildBaseQueryString(List<FuzzyPicklist__mdt> fps, List<Id> recs){
         String obj = recs[0].getSObjectType().getDescribe().getName();
-        String query = 'SELECT Id, Name';
-        if(fps.size()>0){
-            for(FuzzyPicklist__mdt fp : fps){
-                query += ', '+fp.Picklist_Field__r.QualifiedAPIName;
-                query += ', '+fp.Free_Text_Field__r.QualifiedAPIName;
-                if(fp.Match_Last_Evaluated_Date_Time_Field__c != null) {
-                    query += ', '+fp.Match_Last_Evaluated_Date_Time_Field__r.QualifiedAPIName;
-                }
-                if(fp.Text_Last_Populated_Date_Time_Field__c != null) {
-                    query += ', '+fp.Text_Last_Populated_Date_Time_Field__r.QualifiedAPIName;
-                }
-            }
+        // to avoid issues with the same field being used in multiple rules, add to a set to dedupe
+        Set<String> fields = new Set<String> {'Id', 'Name'};
```

Comment:
I know we talked about this. I think this is a better pattern because there could be duplicate fields used across rules so this avoids building a query that won't work.